http.c

```diff
@@ -221,10 +224,55 @@ int http_open(struct http_get *hg, int timeout_ms)
 	return -1;
 }
 
-static int http_write(int fd, const char *buf, int count, int timeout_ms)
+int connection_open(struct connection *conn, struct http_get *hg, int timeout_ms)
+{
+	if (socket_open(hg, timeout_ms))
+		return -IP_ERROR_ERRNO;
+	*conn->fd_ref = hg->fd;
+
+	if(hg->is_https == 0)
+		return IP_ERROR_SUCCESS;
+
+	if(hg->proxy != NULL) {
+		/*
```

Comment:
This suggests that when using http you already return early and the proxy setting is ignored and no warning message is given that the proxy is not supported. (?)